charts/dependency-track/values.yaml

```diff
@@ -0,0 +1,75 @@
+# Default values for dependency-track.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: owasp/dependency-track
+  tag: 3.8.0
+  pullPolicy: IfNotPresent
+
+service:
+  type: ClusterIP
+  port: 80
+
+nfs:
+  server: nfsserveur.your.domain.com
+  path: /nfsserver/dependency-track/data
+
+env:
+  ALPINE_DATABASE_MODE: external
+  ALPINE_DATABASE_URL: jdbc:postgresql://db-postgres:5432/dtrack"
+  ALPINE_DATABASE_DRIVER: org.postgresql.Driver
+  ALPINE_DATABASE_DRIVER_PATH: /extlib/postgresql-42.2.5.jar
+  ALPINE_DATABASE_POOL_ENABLED: true
+  ALPINE_DATABASE_POOL_MAX_SIZE: 10
+  ALPINE_DATABASE_POOL_IDLE_TIMEOUT: 600000
+  ALPINE_DATABASE_POOL_MAX_LIFETIME: 600000
+  ALPINE_LDAP_ENABLED: "true"
+  ALPINE_LDAP_BASEDN: "DC=your,DC=domain,DC=com"
+  ALPINE_LDAP_SECURITY_AUTH: "simple"
+  ALPINE_LDAP_AUTH_USERNAME_FORMAT: "%s"
+  ALPINE_LDAP_ATTRIBUTE_NAME: "sAMAccountName"
+  ALPINE_LDAP_ATTRIBUTE_MAIL: "mail"
+  ALPINE_LDAP_GROUPS_FILTER: "(&(objectClass=group)(objectCategory=Group))"
+  ALPINE_LDAP_USER_GROUPS_FILTER: "(member:1.2.840.113556.1.4.1941:={USER_DN})"
+  ALPINE_LDAP_USER_PROVISIONING: "true"
+  ALPINE_LDAP_TEAM_SYNCHRONIZATION: "true"
+  ALPINE_HTTP_PROXY_ADDRESS: proxy-server.com
+  ALPINE_HTTP_PROXY_PORT: 8080
+
+secretsEnv:
+  # Permet de spécifier la liste des secrets à monter et les clés à injecter en variable d'environnements
+  - secretName: db-dependency-track
+    env:
+      ALPINE_DATABASE_USERNAME: username
+      ALPINE_DATABASE_PASSWORD: password
+  - secretName: ldap-auth
+    env:
+      ALPINE_LDAP_SERVER_URL: LDAP_URL
+      ALPINE_LDAP_BIND_USERNAME: USERNAME
+      ALPINE_LDAP_BIND_PASSWORD: PASSWORD
+  - secretName: proxy-auth
+    env:
+      ALPINE_HTTP_PROXY_USERNAME: username
+      ALPINE_HTTP_PROXY_PASSWORD: password
+
+
+ingress:
+  enabled: true
+  #annotations:
+  #  traefik.ingress.kubernetes.io/redirect-entry-point: https
+  #  traefik.ingress.kubernetes.io/redirect-permanent: "true"
+  path: /
+  hosts:
+   - dependency-track.your.domain.com
+
+resources:
```

Comment:
Par défaut pas de resources, mais on peut les commenter pour proposer les valeurs optimal à l'activation

Reply:
Il reste à commenter le requests CPU

Reply:
Elles l'ont d'abord été selon tes précos jusqu'à ce que le build tombe en fail au lancement de l'application dans le conteneur pour la raison suivante (tirée de l'historique du run https://github.com/cnieg/helm-charts/runs/945432575) :

15:20:05.196 ERROR [RequirementsVerifier] Dependency-Track requires a minimum of 2 CPU cores. Cannot continue.

La conf par défaut de l'environnement de build de Github est la suivante (un peu plus haut que l'erreur dans les logs du run) :
 15:20:05.164 INFO [Config] OS Name:      Linux
15:20:05.165 INFO [Config] OS Version:   5.3.0-1032-azure
15:20:05.165 INFO [Config] OS Arch:      amd64
15:20:05.166 INFO [Config] CPU Cores:    1
15:20:05.166 INFO [Config] Java Vendor:  AdoptOpenJDK
15:20:05.166 INFO [Config] Java Version: 8.0.222

D'où l'imposition (progreesive) de 2 CPUs avec les commits :
- values.yaml: dependency track asks for 2 cpu cores at container init
- values: forgot to drop empty set brackets at explicit resources specs
- values.yaml: github build system is angry about indents (5 instead of 4)

Mais le kube mis à disposition par la plateforme Github ne l'entend pas comme ça (https://github.com/cnieg/helm-charts/runs/948366283) :
Warning  FailedScheduling  59s (x7 over 8m20s)  default-scheduler  0/1 nodes are available: 1 Insufficient cpu.

C'est à partir de là que j'ai commencé à travailler sur notre infra, où ça passe vraisemblablement.